generators/app/index.js

```diff
@@ -1,49 +1,91 @@
-const fs = require('fs')
-const colors = require('colors')
-const Generator = require('yeoman-generator')
-const upath = require('upath')
-const appSettings = require('../../package.json')
+const fs = require('fs');
+const Generator = require('yeoman-generator');
+const upath = require('upath');
+const header = require('./header');
 
 module.exports = class extends Generator {
+
   constructor (args, opts) {
-    super(args, opts)
-    const year = new Date().getYear() + 1900
-    this.log('/------------------------------------------/'.cyan)
-    this.log('/'.cyan + '        SYMPHONY GENERATOR  '.bold + appSettings.version.bold + '         /'.cyan)
-    this.log('/    by platformsolutions@symphony.com     /'.cyan)
-    this.log(`/ (c) ${year} Symphony Communication Services /`.cyan)
-    this.log('/------------------------------------------/'.cyan)
+    super(args, opts);
   }
 
-  init () {
-
+  /**
+   * Displays the Welcome header (current version, year, etc.)
+   */
+  initializing () {
+    header.display(this.log);
   }
 
+  /**
+   * Displays the initial menu (application type, language, auth, etc.)
+   */
   prompting () {
-    return this.prompt([
+
+    const modules = {
+      application: {
+        node: {
+          path: '../node-ext-apps'
+        },
+        java: {
+          path: '../java-ext-apps'
+        }
+      },
+      bot: {
+        node: {
+          path: '../node-bots'
+        },
+        java: {
+          path: '../java'
+        },
+        dotnet: {
+          path: '../dotnet-bots'
+        },
+        python: {
+          path: '../python-bots'
+        }
+      }
+    };
+
+    const projectType = {
+      bot: 'bot',
+      application: 'application',
+
+      all: [ 'bot', 'application' ]
+    };
+
+    const language = {
+      java: 'Java',
+      node: 'Node.js',
+      dotnet: '.Net',
+      python: 'Python',
+
+      all: [ 'Java', 'Node.js', '.Net', 'Python' ]
```

Comment:
I would recommend against define all the properties and the `all` like you did. I find it too error-prone, not future-proof.
If you want to be more future-proof you can do:
```
    const language = {
      java: 'Java',
      node: 'Node.js',
      dotnet: '.Net',
      python: 'Python'
}
Object.defineProperty(language, 'all', { get: function() { return Object.values(this)}, enumerable: false})
```
But more generally, I would not use that property at all. I would just do `Object.values(language)` whenever I need to use the values.

Reply:
Done